src/stcal/resample/resample.py

```diff
@@ -0,0 +1,1660 @@
+                        "pixel_scale_ratio"
+                    )
+                    self._output_wcs = output_wcs.get("wcs")
+                else:
+                    self._output_wcs = output_wcs
+
+                self.check_output_wcs(self._output_wcs)
+
+        else:
+            self.validate_output_model(
+                output_model=output_model,
+                accumulate=accumulate,
+                enable_ctx=enable_ctx,
+                enable_var=enable_var,
+            )
+            self._output_model = output_model
+            self._output_wcs = output_model["wcs"]
+            self._output_pixel_scale = output_model.get("pixel_scale")
+            if output_wcs:
+                log.warning(
+                    "'output_wcs' will be ignored. Using the 'wcs' supplied "
+                    "by the 'output_model' instead."
+                )
+
+        if self._output_pixel_scale is None:
+            self._output_pixel_scale = 3600.0 * np.rad2deg(
+                math.sqrt(
+                    self.get_output_model_pixel_area({"wcs": self._output_wcs})
+                )
+            )
+            log.info(
+                "Computed output pixel scale: "
+                f"{self._output_pixel_scale} arcsec."
+            )
+        else:
+            log.info(
+                f"Output pixel scale: {self._output_pixel_scale} arcsec."
+            )
+
+        self._output_array_shape = self._output_wcs.array_shape
+
+        # Check that the output data shape has no zero-length dimensions
+        npix = np.prod(self._output_array_shape)
+        if not npix:
+            raise ValueError(
+                "Invalid output frame shape: "
+                f"{tuple(self._output_array_shape)}"
+            )
+
+        log.info(f"Driz parameter kernel: {self.kernel}")
+        log.info(f"Driz parameter pixfrac: {self.pixfrac}")
+        log.info(f"Driz parameter fillval: {self.fillval}")
+        log.info(f"Driz parameter weight_type: {self.weight_type}")
+        log.debug(
+            f"Output mosaic size (nx, ny): {self._output_wcs.pixel_shape}"
+        )
+
+        # set up an empty output model (don't allocate arrays at this time):
+        if self._output_model is None:
+            self._output_model = self.create_output_model()
+
+        self.reset_arrays(reset_output=False, n_input_models=n_input_models)
+
+    def get_input_model_pixel_area(self, model):
+        """
+        Computes or retrieves pixel area of an input model. Currently,
+        this is the average pixel area of input model's pixels within either
+        the bounding box (if available) or the entire data array.
+
+        This value is used to compute a scale factor that will be applied
+        to input image data. This scale factor takes into account the
+        difference in the definition of the pixel area reported in model's
+        ``meta`` and the pixel area at the location used to construct
```

Comment:
```suggestion
        difference in the definition of the pixel area reported in
        ``model.meta`` and the pixel area at the location used to construct
```
...although, now that this is written in such a way that everything is passed in and out with dictionaries, is `meta` still relevant?

Reply:
"model: A dictionary containing data arrays and other meta attributes". So I think it is ok to refer to "meta" but not ``model.meta``.

Reply:
I will restore previous text.